src/modules/analysis/fpsigmatcher/Utils.c

```diff
@@ -110,8 +110,9 @@ char ** get_filenames(const char * directory, int * num_of_files) {
 	while ( pDirEnt != NULL ) {
 		/* Skip hidden entries starting with dot */
 		if( pDirEnt->d_name[0] != '.' ) {
+			int fileNameSize = snprintf(filename, sizeof(filename), "%s/%s", directory, pDirEnt->d_name);
```

Comment:
I dont really get this change: Was there something wrong with the implementation or should this make the code easier to understand?

Reply:
snprintf returns negative values on failing, which are smaller than sizeof(filename). So the return value should be in the range (0, sizeof(filename)) when it is successful.

Reply:
That makes sense!

Reply:
In that case shouldnt we additionally skip if `fileNameSize < 0` ?

Reply:
we skip them if fileNameSize is not > 0. I was thinking that 0 cannot be good either.

Reply:
You're right, I was confused by all the parentheses.